Header file not found when running `assembleRelease`

It works on my macOS device but Linux CI.

- Reproducer: https://github.com/Goooler/MihomoForAndroid/pull/100
- Full stacktrace: https://productionresultssa9.blob.core.windows.net/actions-results/2caca65f-3c23-43f5-8cc6-fd6b48261dc8/workflow-job-run-0c11c28a-1415-522b-ae5b-ebac984e8915/logs/job/job-logs.txt?rsct=text%2Fplain&se=2026-04-22T02%3A49%3A33Z&sig=%2BrlMRxYTjLklzOAPKO5Lt1ikXm3Ugogcu5xmgIFJZr4%3D&ske=2026-04-22T06%3A14%3A06Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2026-04-22T02%3A14%3A06Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-11-05&sp=r&spr=https&sr=b&st=2026-04-22T02%3A39%3A28Z&sv=2025-11-05

```
4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:buildCMakeRelWithDebInfo[arm64-v8a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `/home/runner/work/MihomoForAndroid/MihomoForAndroid/core/.cxx/RelWithDebInfo/5j636dh2/arm64-v8a'
  [1/4] Building C object CMakeFiles/bridge.dir/main.c.o
  FAILED: CMakeFiles/bridge.dir/main.c.o
  /usr/local/lib/android/sdk/ndk/29.0.14206865/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=aarch64-none-linux-android28 --sysroot=/usr/local/lib/android/sdk/ndk/29.0.14206865/toolchains/llvm/prebuilt/linux-x86_64/sysroot -Dbridge_EXPORTS -I/home/runner/work/MihomoForAndroid/MihomoForAndroid/core/build/intermediates/go/Release/arm64-v8a -I/home/runner/work/MihomoForAndroid/MihomoForAndroid/core/src/main/golang/native -I/home/runner/work/MihomoForAndroid/MihomoForAndroid/core/.cxx/RelWithDebInfo/5j636dh2/arm64-v8a -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -O2 -g -DNDEBUG -fPIC -MD -MT CMakeFiles/bridge.dir/main.c.o -MF CMakeFiles/bridge.dir/main.c.o.d -o CMakeFiles/bridge.dir/main.c.o -c /home/runner/work/MihomoForAndroid/MihomoForAndroid/core/src/main/cpp/main.c
  /home/runner/work/MihomoForAndroid/MihomoForAndroid/core/src/main/cpp/main.c:7:10: fatal error: 'libclash.h' file not found
      7 | #include "libclash.h"
        |          ^~~~~~~~~~~~
  1 error generated.
  [2/4] Building C object CMakeFiles/bridge.dir/bridge_helper.c.o
  [3/4] Building C object CMakeFiles/bridge.dir/jni_helper.c.o
  ninja: build stopped: subcommand failed.

  C++ build system [build] failed while executing:
      /usr/local/lib/android/sdk/cmake/4.1.2/bin/ninja \
        -C \
        /home/runner/work/MihomoForAndroid/MihomoForAndroid/core/.cxx/RelWithDebInfo/5j636dh2/arm64-v8a \
        bridge
    from /home/runner/work/MihomoForAndroid/MihomoForAndroid/core

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 26s
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -33,11 +33,11 @@ jobs:
       - run: ./gradlew spotlessCheck
       - name: Build
         run: |
-          ./gradlew app:assembleMetaRelease
+          ./gradlew app:assembleRelease
       - name: Delete release if exist then create release
         if: github.repository == 'Goooler/MihomoForAndroid' && github.event_name == 'push'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           gh release view nightly && gh release delete nightly -y --cleanup-tag
-          gh release create nightly "./app/build/outputs/apk/meta/release/"*.apk --prerelease --generate-notes
+          gh release create nightly "./app/build/outputs/apk/release/"*.apk --prerelease --generate-notes

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -30,10 +30,10 @@ jobs:
       - uses: gradle/actions/setup-gradle@v6
       - name: Build
         run: |
-          ./gradlew app:assembleMetaRelease
+          ./gradlew app:assembleRelease
       - name: Create release
         if: github.repository == 'Goooler/MihomoForAndroid'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          gh release create ${{ github.ref_name }} "./app/build/outputs/apk/meta/release/"*.apk --generate-notes
+          gh release create ${{ github.ref_name }} "./app/build/outputs/apk/release/"*.apk --generate-notes

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -37,7 +37,7 @@ Run checks before submitting a pull request:
 
 ```bash
 ./gradlew spotlessCheck
-./gradlew app:assembleMetaRelease
+./gradlew app:assembleRelease
 ```
 
 ### Pull request guidelines

--- a/README.md
+++ b/README.md
@@ -34,10 +34,8 @@ Feature of [Clash.Meta](https://github.com/MetaCubeX/mihomo)
 4. (Optional) Custom app package name. Add the following configuration to `local.properties`.
 
    ```properties
-   # config your ownn applicationId, or it will be 'com.github.metacubex.clash'
+   # config your own applicationId, or it will be 'com.github.metacubex.clash'
    custom.application.id=com.my.compile.clash
-   # remove application id suffix, or the applicaion id will be 'com.github.metacubex.clash.alpha'
-   remove.suffix=true
 
 5. Create `signing.properties` in project root with
 
@@ -51,12 +49,13 @@ Feature of [Clash.Meta](https://github.com/MetaCubeX/mihomo)
 6. Build
 
    ```bash
-   ./gradlew app:assembleMetaRelease
+   ./gradlew app:assembleRelease
    ```
 
 ### Automation
 
-APP package name is `com.github.metacubex.clash.meta`
+APP package name is `com.github.metacubex.clash`
+Automation action strings remain `com.github.metacubex.clash.meta.action.*` for compatibility.
 
 - Toggle Clash.Meta service status
   - Send intent to activity `com.github.kr328.clash.ExternalControlActivity` with action `com.github.metacubex.clash.meta.action.TOGGLE_CLASH`

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -16,34 +16,9 @@ android {
     versionName = "2.12.1"
     resValue("integer", "release_code", versionCode.toString())
     resValue("string", "release_name", "v$versionName")
+    resValue("string", "launch_name", "@string/launch_name_meta")
+    resValue("string", "application_name", "@string/application_name_meta")
     ndk.abiFilters += listOf("arm64-v8a", "armeabi-v7a", "x86", "x86_64")
-  }
-
-  val localProperties = Properties()
-  val localPropertiesFile = rootProject.file("local.properties")
-  if (localPropertiesFile.exists()) {
-    localPropertiesFile.inputStream().use(localProperties::load)
-  }
-  val removeSuffix = localProperties.getProperty("remove.suffix")?.toBoolean() == true
-
-  productFlavors {
-    named("alpha") {
-      resValue("string", "launch_name", "@string/launch_name_alpha")
-      resValue("string", "application_name", "@string/application_name_alpha")
-      if (!removeSuffix) {
-        applicationIdSuffix = ".alpha"
-        versionNameSuffix = ".Alpha"
-      }
-    }
-    named("meta") {
-      isDefault = true
-      resValue("string", "launch_name", "@string/launch_name_meta")
-      resValue("string", "application_name", "@string/application_name_meta")
-      if (!removeSuffix) {
-        applicationIdSuffix = ".meta"
-        versionNameSuffix = ".Meta"
-      }
-    }
   }
 
   val keystore = rootProject.file("signing.properties")
@@ -97,7 +72,7 @@ androidComponents {
         val abiName =
           output.filters.find { it.filterType == FilterConfiguration.FilterType.ABI }?.identifier
             ?: "universal"
-        val newApkName = "cmfa-${versionName.get()}-meta-$abiName-${variant.buildType}.apk"
+        val newApkName = "cmfa-${versionName.get()}-$abiName-${variant.buildType}.apk"
         outputFileName = newApkName
       }
     }

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -22,15 +22,6 @@ allprojects {
       compileSdk = 37
       defaultConfig.apply { minSdk = 28 }
       ndkVersion = "29.0.14206865"
-      flavorDimensions += "feature"
-      productFlavors {
-        create("alpha") { dimension = "feature" }
-        create("meta") { dimension = "feature" }
-      }
-      sourceSets {
-        getByName("meta") { java.directories.add("src/foss/java") }
-        getByName("alpha") { java.directories.add("src/foss/java") }
-      }
       compileOptions.apply {
         sourceCompatibility(libs.versions.jvmTarget.get())
         targetCompatibility(libs.versions.jvmTarget.get())

--- a/core/build.gradle.kts
+++ b/core/build.gradle.kts
@@ -39,6 +39,5 @@ androidComponents.onVariants { variant ->
     ?.addAll(
       "-DGO_SOURCE:STRING=$golangSource",
       "-DGO_OUTPUT:STRING=${goBaseOutputDir.get().asFile}",
-      "-DFLAVOR_NAME:STRING=${variant.flavorName.orEmpty()}",
     )
 }

--- a/core/src/main/cpp/CMakeLists.txt
+++ b/core/src/main/cpp/CMakeLists.txt
@@ -4,8 +4,8 @@ cmake_minimum_required(VERSION 4.0)
 
 message(STATUS "CMAKE_CURRENT_SOURCE_DIR= ${CMAKE_CURRENT_SOURCE_DIR}")
 execute_process(
-  COMMAND git submodule foreach git log -1 --format=%H   
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} 
+  COMMAND git submodule foreach git log -1 --format=%H
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   OUTPUT_VARIABLE COMMIT_HASH
 )
 string(REPLACE "\n" ";" COMMIT_HASH "${COMMIT_HASH}")
@@ -17,7 +17,7 @@ message(STATUS "git hash= ${COMMIT_HASH}")
 # 获取分支名称
 execute_process(
   COMMAND git submodule foreach git branch -r --contains ${COMMIT_HASH}
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} 
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   OUTPUT_VARIABLE CURRENT_BRANCH
 )
 string(REPLACE "\n" ";" CURRENT_BRANCH "${CURRENT_BRANCH}")
@@ -48,14 +48,14 @@ project(clash-bridge C)
 set(CMAKE_POSITION_INDEPENDENT_CODE on)
 set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3")
 
-set(GO_OUTPUT_BASE ${GO_OUTPUT}/${FLAVOR_NAME})
+set(GO_OUTPUT_BASE ${GO_OUTPUT}/)
 
 if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}Debug")
+    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}debug")
 elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
-    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}Release")
+    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}release")
 elseif ("${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo")
-    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}Release")
+    set(GO_OUTPUT_BASE "${GO_OUTPUT_BASE}release")
 else ()
     message(FATAL_ERROR "Unknown build type ${CMAKE_BUILD_TYPE}")
 endif ()

